app/src/main/kotlin/com/exallium/todoapp/editnote/EditNotePresenter.kt

```diff
@@ -13,35 +14,48 @@ import timber.log.Timber
  * Presenter for Edit Note Screen
  */
 class EditNotePresenter(view: EditNoteView,
-                          private val model: EditNoteModel,
-                          private val screenBundleHelper: ScreenBundleHelper) : BasePresenter<EditNoteView>(view) {
+                        private val model: EditNoteModel,
+                        private val screenBundleHelper: ScreenBundleHelper,
+                        private val idFactory: IdFactory) : BasePresenter<EditNoteView>(view) {
 
-    private val showNewNoteDetailSubscriberFn: (Unit?) -> (Unit) = {
-        view.showNewNoteDetail(getArgs())
+    private val showAllNotesSubscriberFn: (Unit?) -> (Unit) = {
+        view.showAllNotes(getArgs())
     }
 
-    override fun onViewCreated() {
-        screenBundleHelper.setTitle(getArgs(), R.string.edit_note_screen_title)
-        val noteId: String = screenBundleHelper.getNoteId(getArgs())
-
-        setupGetNoteDetailSubscription(noteId)
-
-        setupSaveNoteSubscription(noteId)
+    private val showNoteDetailSubscriberFn: (Unit?) -> (Unit) = {
+        view.showNoteDetail(getArgs())
+    }
 
-        view.cancelEditNoteClicks().map { null }.subscribe(showNewNoteDetailSubscriberFn).addToComposite()
+    override fun onViewCreated() {
+        val args = getArgs()
+        if (screenBundleHelper.getNoteId(args) == null) {
+            screenBundleHelper.setTitle(args, R.string.create_note_screen_title)
+            setupSaveNewNoteSubscription()
+            setupCancelButtonClicksSubscription { showAllNotesSubscriberFn }
+        } else {
+            screenBundleHelper.setTitle(args, R.string.edit_note_screen_title)
+            val noteId: String = screenBundleHelper.getNoteId(args)!!
+            setupGetNoteDetailSubscription(noteId)
+            setupSaveEditedNoteSubscription(noteId)
+            setupCancelButtonClicksSubscription { showNoteDetailSubscriberFn }
+        }
+        setupTextChangedSubscription()
+    }
 
-        setupTextViewsChanged()
+    fun setupCancelButtonClicksSubscription(cancelButtonFn: (Unit?) -> (Unit)) {
+        cancelButtonFn
```

Comment:
i realized that it's wrong. how do you pass a subscriber FN as a param to a method so for create setupCancelButtonClicksSubscription would do 

`        view.cancelEditNoteClicks().map { null }.subscribe(showAllNotesSubscriberFn).addToComposite()`

and for edit screen, it would use showNoteDetailSubscriberFn